src/base/ceramicApiKit.ts

```diff
@@ -0,0 +1,197 @@
+                      transactionHash: "${safeTxHash}"
+                      confirmationsRequired: ${threshold}
+                  }
+                  clientMutationId: null
+              }
+          ) {
+              clientMutationId
+          }
+      }
+    `);
+
+
+    return Promise.resolve();
+  }
+
+  async getTransaction(safeTxHash: string): Promise<SafeMultisigTransactionResponse> {
+
+    try {
+      if (!this.composeClient) {
+        await this.connect();
+      }
+      const transactionIndex = await this.composeClient.executeQuery(`
+          query TransactionIndex {
+              transactionIndex(
+                  first: 1
+                  filters: { where: { safeTxHash: { equalTo: "${safeTxHash}" } } }
+              ) {
+                  edges {
+                      cursor
+                      node {
+                          id
+                          to
+                          data
+                          safe
+                          nonce
+                          value
+                          proposer
+                          operation
+                          safeTxHash
+                          signatures
+                          submissionDate
+                          transactionHash
+                          confirmationsRequired
+                      }
+                  }
+              }
+          }
+        `);
+
+      const confirmationsIndex: ConfirmationIndexData = await this.composeClient.executeQuery(`
+            query ConfirmationIndex {
+                confirmationIndex(
+                    first: 10
+                    filters: { where: { transactionHash: { equalTo: "${safeTxHash}" } } }
+                ) {
+                    edges {
+                        node {
+                            owner
+                            id
+                            signature
+                            signatureType
+                            submissionDate
+                            transactionHash
+                            confirmationType
+                        }
+                    }
+                }
+            }
+        `) as ConfirmationIndexData;
+      const confirmations = confirmationsIndex.data.confirmationIndex.edges.map((edge) => edge.node).filter((confirmation) => {
+        const { signature } = confirmation;
+        const r = signature.slice(0, 66);
+        const s = `0x${signature.slice(66, 130)}`;
+        let v = parseInt(signature.slice(130, 132), 16);
+        if (v !== 27 && v !== 28) {
+          v = 27;
+        }
+        // normalize signature
+        const normalizedSignature = r + s.slice(2) + (v).toString(16).padStart(2, '0');
+        return ethers.verifyMessage(ethers.getBytes(safeTxHash), normalizedSignature).toLowerCase() === confirmation.owner.toLowerCase();
+      })
+      return {
+        ...transactionIndex.data.transactionIndex.edges[0].node,
```

Comment:
How to make sure the txdata is correct?

Reply:
validation would be the only answer, but this just reminds me of attack proof
what we can do if there are massive false data with the same `safeTxHash `, how to query the right one record 🤔

Reply:
I don't think we need to store the data on top of the remote server, we just need to generate it locally and compare it with safeTxHash, the current way of doing this is to get the first data in the remote storage, but there is no trustworthy source for this data

Reply:
nice comment, 🤔 very interesting idea，we may just drop the data model `Transaction` and only use `Confirmation`, let me double confirm does the idea work

Reply:
I have done refactor, now it is reading TxData from local generation
@xiaoch05 and would like to know your suggestion, what do you think, let's remove Ceramic Transaction Model for good, or keep it, for potential later use?